pkg/promlib/querydata/response.go

```diff
@@ -113,14 +113,16 @@ func addMetadataToMultiFrame(q *models.Query, frame *data.Frame) {
 	}
 	frame.Fields[0].Config = &data.FieldConfig{Interval: float64(q.Step.Milliseconds())}
 
-	customName := getName(q, frame.Fields[1])
-	if customName != "" {
-		frame.Fields[1].Config = &data.FieldConfig{DisplayNameFromDS: customName}
-	}
+	if frame.Meta.Type != "heatmap-cells" {
```

Comment:
This condition can return, and then the following code can be dedented.

Reply:
Please recheck the code. I updated it.